tests/e2e/p/proposed_validators.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/tests/fixture/e2e"
+	"github.com/ava-labs/avalanchego/tests/fixture/tmpnet"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/platformvm"
+)
+
+var _ = e2e.DescribePChain("[Proposed Validators]", func() {
+	var (
+		tc      = e2e.NewTestContext()
+		require = require.New(tc)
+	)
+
+	ginkgo.It("should be able to fetch proposed validators", func() {
+		tc.By("creating a new private network to ensure isolation from other tests")
+		privateNetwork := tmpnet.NewDefaultNetwork("avalanchego-e2e-proposed-validators")
+		// Initialize config flags
+		privateNetwork.DefaultFlags = tmpnet.FlagsMap{}
+		// Ensure that the proposer uses the current height
```

Comment:
Yes. I went with a private network with `config.ProposerVMUseCurrentHeightKey` explicitly set so that this test is preserved in case the default tmpnet settings ever change. Happy to ditch the private network altogether if you think that's overkill.

Reply:
That's a much better approach. I put the suggested code into the existing `confirming proposed validators are the same as current validators` spec, since it relies on this config being set correctly.